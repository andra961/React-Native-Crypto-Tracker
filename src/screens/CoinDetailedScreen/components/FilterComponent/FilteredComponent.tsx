import React, {memo} from 'react';
import {View, Text, Pressable} from 'react-native';
import COLORS from '../../../../constants/colors';

const FilteredComponent = ({
  filterDay,
  filterText,
  selectedRange,
  setSelectedRange,
}: {
  filterDay: string;
  filterText: string;
  selectedRange: string;
  setSelectedRange: (arg: string) => void;
}) => {
  const isFilterSelected = (filter: string) => filter === selectedRange;

  return (
    <Pressable
      style={{
        paddingHorizontal: 10,
        paddingVertical: 5,
        borderRadius: 5,
        backgroundColor: isFilterSelected(filterDay)
          ? COLORS.GREY
          : 'transparent',
      }}
      onPress={() => setSelectedRange(filterDay)}>
      <Text
        style={{
          color: isFilterSelected(filterDay)
            ? COLORS.PRIMARY
            : COLORS.DEFAULT_GREY,
        }}>
        {filterText}
      </Text>
    </Pressable>
  );
};

export default memo(FilteredComponent);
