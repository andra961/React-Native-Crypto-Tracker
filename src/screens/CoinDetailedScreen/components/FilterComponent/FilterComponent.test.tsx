import 'react-native';
import React from 'react';
import FilteredComponent from './';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import {render, fireEvent} from '@testing-library/react-native';

describe('Filtered Component test(renderer)', () => {
  it('should render correctly', () => {
    renderer.create(
      <FilteredComponent
        filterDay=""
        filterText=""
        selectedRange=""
        setSelectedRange={() => {}}
      />,
    );
  });
});

describe('Filtered Component test(testing-library)', () => {
  it('should render correctly', () => {
    const mockFn = jest.fn();

    const filtered = render(
      <FilteredComponent
        filterDay="28"
        filterText="28D"
        selectedRange="28"
        setSelectedRange={mockFn}
      />,
    );

    const text = filtered.getByText('28D');

    expect(text).not.toBeNull();

    fireEvent.press(text);

    expect(mockFn).toBeCalledWith('28');

    expect(filtered).toMatchSnapshot();
  });
});
