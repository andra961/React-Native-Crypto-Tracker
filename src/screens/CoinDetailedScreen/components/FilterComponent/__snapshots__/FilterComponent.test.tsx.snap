// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filtered Component test(testing-library) should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#1e1e1e",
      "borderRadius": 5,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
  >
    28D
  </Text>
</View>
`;
